### PR TITLE
[5.0] TransfomationMonitor use BaseRequestHandler

### PR DESCRIPTION
**Pull request by TaykYoku**
issue https://github.com/DIRACGrid/WebAppDIRAC/issues/658

BEGINRELEASENOTES

CHANGE: TransformationMonitor use BaseRequestHandler

ENDRELEASENOTES

### CODE REVIEW

File: src/WebAppDIRAC/WebApp/handler/TransformationMonitorHandler.py [hunk 1]
```diff
@@ -8,512 +8,417 @@
...
+        if not result["OK"]:
+            return json.dumps({"success": "false", "error": result["Message"]})
+
+        data = result["Value"]
+
+        if "TotalRecords" not in data:
+            return json.dumps({"success": "false", "result": "", "error": "Data structure is corrupted"})
+
+        if (total := data["TotalRecords"]) < 1:
+            return json.dumps({"success": "false", "result": "", "error": "There were no data matching your selection"})
```
Comment (chrisburr) [inline src/WebAppDIRAC/WebApp/handler/TransformationMonitorHandler.py]: Is the `json.dumps` needed? Or could the dictionary be returned directly?

File: src/WebAppDIRAC/WebApp/handler/TransformationMonitorHandler.py [hunk 2]
```diff
@@ -8,512 +8,417 @@
...
+                                "result": callback,
+                                "total": total,
+                                "extra": data["Extras"],
+                                "date": timestamp,
+                            }
+                        return {"success": "true", "result": callback, "total": total, "date": timestamp}
+                    return {"success": "false", "result": "", "error": "There are no data to display"}
+                return {"success": "false", "result": "", "error": "ParameterNames field is undefined"}
+            return {"success": "false", "result": "", "error": "Data structure is corrupted"}
+        return {"success": "false", "result": "", "error": "There were no data matching your selection"}
```
Comment (chrisburr) [inline src/WebAppDIRAC/WebApp/handler/TransformationMonitorHandler.py]: I'm ignoring most of style comments but I think this function is bad enough that we should fix it:

```suggestion
        total = len(data.get("TotalRecords", []))
        if total == 0:
            return {"success": "false", "result": "", "error": "There were no data matching your selection"}
        if not ("ParameterNames" in data and "Records" in data):
            return {"success": "false", "result": "", "error": "Data structure is corrupted"}
        head = data.get("ParameterNames", [])
        if len(head) == 0:
            return {"success": "false", "result": "", "error": "ParameterNames field is undefined"}
        if len(data["Records"]) == 0:
            return {"success": "false", "result": "", "error": "There are no data to display"}
        callback = []
        for job in data["Records"]:
            callback.append(dict(zip(head, job)))
        timestamp = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M [UTC]")
        if "Extras" in data:
            return {
                "success": "true",
                "result": callback,
                "total": total,
                "extra": data["Extras"],
                "date": timestamp,
            }
        return {"success": "true", "result": callback, "total": total, "date": timestamp}     
```

Comment (chrisburr) [reply src/WebAppDIRAC/WebApp/handler/TransformationMonitorHandler.py]: Also I think we should be a little more selective with the use of the `:=` operator, things like this are quite tricky to parse:

```
if (headLength := len(head := data["ParameterNames"])) > 0:
```

File: src/WebAppDIRAC/WebApp/handler/TransformationMonitorHandler.py (hunk 1 above)
Comment (TaykYoku) [reply src/WebAppDIRAC/WebApp/handler/TransformationMonitorHandler.py]: No, `BaseRequestHandler` already has encoder. Fixed

**Pull request merged**